File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

/**
 * 品牌相关接口
 *
 * @author Administrator
 */
public interface BrandService {

    /**
     * 获取品牌列表
     *
     * @return java.util.List<com.pinyougou.pojo.TbBrand>
     */
    List<TbBrand> findAll();

    /**
     * 分页查询品牌列表
     *
     * @param pageNum  当前页码
     * @param pageSize 每页记录数
     * @return entity.PageResult
     */
    PageResult findByPage(int pageNum, int pageSize);

    /**
     * 新增品牌
     *
     * @param brand 品牌对象
     */
    void add(TbBrand brand);

    /**
     * 按 id 获取品牌
     *
     * @param id 品牌id
     * @return com.pinyougou.pojo.TbBrand
     */
    TbBrand findOne(long id);

    /**
     * 更新品牌
     *
     * @param brand 要更新的品牌
     */
    void update(TbBrand brand);

    /**
     * 删除品牌
     *
     * @param ids 被删除品牌的品牌id
     */
    void delete(long[] ids);

    /**
     * 按条件查询品牌列表
     *
     * @param brand 查询条件
     * @param page  当前页码
     * @param size  每页显示记录数
     * @return entity.PageResult
     */
    PageResult findByPage(TbBrand brand, int page, int size);

    /**
     * 返回品牌下拉列表数据
     *
     * @return java.util.List<java.util.Map>
     */
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//自定义服务
app.service('brandService', function ($http) {

    // 分页查询品牌列表
    this.findByPage = function (page, size) {
        return $http.get('../brand/findByPage.do?page=' + page + '&size=' + size);
    };

    // 新增品牌
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    };

    // 按id查询品牌
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };

    // 修改品牌
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    };

    // 删除品牌
    this.dele = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    }

    // 查询品牌
    this.search = function (page,size,queryParam) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size, queryParam);
    }

    // 获取品牌下拉列表需要的数据集
    this.selectOptionList = function () {
        return $http.get('../brand/selectOptionList.do');
    }

});